code/modules/research/nanites/public_chamber.dm

```diff
@@ -0,0 +1,173 @@
+
+	var/locked_state = locked
+	locked = TRUE
+
+	//TODO OMINOUS MACHINE SOUNDS
+	set_busy(TRUE, "[initial(icon_state)]_raising")
+	addtimer(CALLBACK(src, .proc/set_busy, TRUE, "[initial(icon_state)]_active"),20)
+	addtimer(CALLBACK(src, .proc/set_busy, TRUE, "[initial(icon_state)]_falling"),60)
+	addtimer(CALLBACK(src, .proc/complete_injection, locked_state),80)
+
+/obj/machinery/public_nanite_chamber/proc/complete_injection(locked_state)
+	//TODO MACHINE DING
+	locked = locked_state
+	set_busy(FALSE)
+	if(!occupant)
+		return
+	occupant.AddComponent(/datum/component/nanites, 75, cloud_id)
+
+/obj/machinery/public_nanite_chamber/update_icon()
+	cut_overlays()
+
+	if(!(stat & (NOPOWER|BROKEN)))
+		if(busy || locked)
+			add_overlay("red")
+			if(locked)
+				add_overlay("bolted")
+		else
+			add_overlay("green")
+
+	//TODO make an overlay for the panel
+	if((stat & MAINT) || panel_open)
+		icon_state = initial(icon_state)+ (state_open ? "_open" : "") + "_maintenance"
+		return
+
+	//running and someone in there
+	if(occupant)
+		if(busy)
+			icon_state = busy_icon_state
+		else
+			icon_state = initial(icon_state)+ "_occupied"
+		return
+
+	//running
+	icon_state = initial(icon_state)+ (state_open ? "_open" : "")
+
+/obj/machinery/public_nanite_chamber/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/public_nanite_chamber/proc/toggle_open(mob/user)
+	if(panel_open)
+		to_chat(user, "<span class='notice'>Close the maintenance panel first.</span>")
+		return
+
+	if(state_open)
+		close_machine()
+		return
+
+	else if(locked)
+		to_chat(user, "<span class='notice'>The bolts are locked down, securing the door shut.</span>")
+		return
+
+	open_machine()
+
+/obj/machinery/public_nanite_chamber/container_resist(mob/living/user)
+	if(!locked)
+		open_machine()
+		return
+	if(busy)
+		return
+	user.changeNext_move(CLICK_CD_BREAKOUT)
+	user.last_special = world.time + CLICK_CD_BREAKOUT
+	user.visible_message("<span class='notice'>You see [user] kicking against the door of [src]!</span>", \
+		"<span class='notice'>You lean on the back of [src] and start pushing the door open... (this will take about [DisplayTimeText(breakout_time)].)</span>", \
+		"<span class='italics'>You hear a metallic creaking from [src].</span>")
+	if(do_after(user,(breakout_time), target = src))
+		if(!user || user.stat != CONSCIOUS || user.loc != src || state_open || !locked || busy)
+			return
+		locked = FALSE
+		user.visible_message("<span class='warning'>[user] successfully broke out of [src]!</span>", \
+			"<span class='notice'>You successfully break out of [src]!</span>")
+		open_machine()
+
+/obj/machinery/public_nanite_chamber/close_machine(mob/living/carbon/user)
+	if(!state_open)
+		return FALSE
+
+	..(user)
+
+	. = TRUE
+
+	if(occupant)
+		var/mob/living/L = occupant
+		var/list/nanite_data = list()
+		SEND_SIGNAL(L, COMSIG_NANITE_GET_DATA, nanite_data)
+		if(!LAZYLEN(nanite_data) || nanite_data["cloud_id"] == cloud_id) //can be used to switch cloud id
+			return
```

Comment:
The signal placement is fine here but you want to use a signal return flag rather than checking the data yourself
```
if(SEND_SIGNAL(L, COMSIG_NANITE_INJECTING, user) & COMPONENT_BLOCK_INJECTION)
  return
```
Check the relevant args in the component itself